e2e-tests/playwright/docs/accessibility/interaction_testing.md

```diff
@@ -0,0 +1,405 @@
+    await expect(locator).toBeVisible();
+    await expect(locator).toBeFocused();
+
+    // Check if element has focus-visible state
+    const hasFocusVisible = await locator.evaluate((element) => element.matches(':focus-visible'));
+
+    expect(hasFocusVisible, 'Element should have focus-visible state').toBe(true);
+}
+
+test('focus indicators visibility', async ({page}) => {
+    const button = page.locator('button').first();
+
+    // Tab to button (should show focus indicator)
+    await page.keyboard.press('Tab');
+    await toBeFocusedWithFocusVisible(button);
+
+    // Click button (may not show focus indicator depending on browser)
+    await button.blur();
+    await button.click();
+    await expect(button).toBeFocused();
+});
+```
+
+### **2. Complex Navigation Patterns**
+
+```typescript
+test('hierarchical menu navigation', async ({page}) => {
+    await page.locator('[data-testid="menu-button"]').focus();
+    await page.keyboard.press('Enter');
+
+    // Navigate to submenu trigger
+    const submenuTrigger = page.getByRole('menuitem', {name: /has submenu/});
+    await submenuTrigger.focus();
+
+    // Open submenu with right arrow
+    await page.keyboard.press('ArrowRight');
+
+    // Verify submenu opened and focused
+    const submenu = page.locator('[role="menu"]').nth(1);
+    await expect(submenu).toBeVisible();
+    await expect(submenu.getByRole('menuitem').first()).toBeFocused();
+
+    // Navigate in submenu
+    await page.keyboard.press('ArrowDown');
+    await expect(submenu.getByRole('menuitem').nth(1)).toBeFocused();
+
+    // Return to parent menu with left arrow
+    await page.keyboard.press('ArrowLeft');
+    await expect(submenu).toBeHidden();
+    await expect(submenuTrigger).toBeFocused();
+});
+```
+
+### **3. Screen Reader Simulation**
+
+```typescript
+test('screen reader content structure', async ({page}) => {
+    // Test heading structure
+    const headings = page.locator('h1, h2, h3, h4, h5, h6');
+    const headingLevels = await headings.evaluateAll((elements) =>
+        elements.map((el) => parseInt(el.tagName.substring(1))),
+    );
+
+    // Verify logical heading hierarchy
+    for (let i = 1; i < headingLevels.length; i++) {
+        const levelDiff = headingLevels[i] - headingLevels[i - 1];
+        expect(levelDiff, 'Heading levels should not skip').toBeLessThanOrEqual(1);
+    }
+
+    // Test landmark navigation
+    const landmarks = page.locator('[role="banner"], [role="navigation"], [role="main"], [role="contentinfo"]');
+    const landmarkCount = await landmarks.count();
+    expect(landmarkCount, 'Page should have proper landmarks').toBeGreaterThan(0);
+
+    // Test skip links
+    await page.keyboard.press('Tab');
+    const firstFocusable = await page.evaluate(() => document.activeElement?.textContent);
+    expect(firstFocusable, 'First focusable should be skip link').toContain('Skip');
+});
+```
+
+## Utility Functions
```

Comment:
I'd expect this section to be a list of helpful utility functions in the code rather than en example of a utility function that we could define